### PR TITLE
`tbot configure` command for assisting Machine ID configuration

### LINKED ISSUE #11483: Improve machine ID config generation
I was reading the machine ID guide:

https://goteleport.com/docs/machine-id/guides/jenkins/

I am wondering if we can reduce the steps with creating a systemd unit and /etc/tbot.yaml 
Most of the content there is a boilerplate. See how event handler does it:

https://goteleport.com/docs/setup/guides/fluentd/#step-26-generate-configuration

```bash
./teleport-event-handler configure . teleport.example.com:443
```

I am wondering if `tbot configure` could generate all necessary files for users in one command

### PR DESCRIPTION
**Pull request by strideynet**
Closes https://github.com/gravitational/teleport/issues/11483

Command outputs YAML to stdout or to a file, allowing the user to create a configuration file from parameters. This assists users with achieving a working tbot configuration as fast as possible.

The `-o` flag can be provided to write the configuration directly to a file, but when this is omitted, the configuration is just output to stdout:

```sh
➜  teleport git:(strideynet/machine-id-yaml-configuration-output) ✗ build/tbot configure -a "foo.com" --token "xyyzz" -o ~/test 
INFO [TBOT]      Generated config file written to file: /Users/noahstride/test tbot/main.go:177
➜  teleport git:(strideynet/machine-id-yaml-configuration-output) ✗ cat ~/test
# tbot config file generated by `configure` command
onboarding:
  token: xyyzz
  ca_path: ""
  ca_pins: []
  join_method: token
storage:
  directory:
    path: /var/lib/teleport/bot
    symlinks: try-secure
    acls: "off"
debug: false
auth_server: foo.com
certificate_ttl: 1h0m0s
renewal_interval: 20m0s
oneshot: false
➜  teleport git:(strideynet/machine-id-yaml-configuration-output) ✗ build/tbot configure -a "foo.com" --token "xyyzz"          
# tbot config file generated by `configure` command
onboarding:
  token: xyyzz
  ca_path: ""
  ca_pins: []
  join_method: token
storage:
  directory:
    path: /var/lib/teleport/bot
    symlinks: try-secure
    acls: "off"
debug: false
auth_server: foo.com
certificate_ttl: 1h0m0s
renewal_interval: 20m0s
oneshot: false
```

## `golden`

This also introduces a test helper package called `golden`. I'm happy to extract this into a separate PR if needs be.  I have found golden files to be a very convenient and effective way of handling large expected outputs from tests in a way that can be easily updated, and inspected in PR diffs. 

"Golden files" are expected test outputs stored as files under `testdata/` rather than embedded directly within the test table. Running the tests with the environment variable `GOLDEN_UPDATE=1` causes this files to be updated, rather than asserted against.

### CODE REVIEW

File: tool/tbot/main_test.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+func Test_onConfigure(t *testing.T) {
+	t.Parallel()
+
+	// This is slightly rubbish, but due to the global nature of `botfs`,
+	// it's difficult to configure the default acl and symlink values to be
+	// the same across dev laptops and GCB.
+	// If we switch to a more dependency injected model for botfs, we can
+	// ensure that the test one returns the same value across operating systems.
+	normalizeOSDependentValues := func(data []byte) []byte {
```
Comment (strideynet) [inline tool/tbot/main_test.go:39]: I'm really not a fan of this, but I didn't want to completely refactor `botfs` to be a dependency injected struct without a bit more investigation.

I'm completely open to some more suggestions here.

File: tool/tbot/main_test.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+		str := string(data)
+		str = strings.ReplaceAll(
+			str, "symlinks: try-secure", "symlinks: secure",
+		)
+		str = strings.ReplaceAll(
+			str, `acls: "off"`, "acls: try",
+		)
+		return []byte(str)
+	}
```
[the remark this answers is not in the document]
Comment (strideynet) [reply tool/tbot/main_test.go:46]: Completely forgot that was a stdlib function 🤦

File: tool/tbot/main_test.go (hunk 1 above)
Comment (timothyb89) [reply tool/tbot/main_test.go:36]: Yeah, that's unfortunate. The overly-simple botfs design has caused some problems already and I'm not opposed to refactoring it.

File: tool/tbot/config/config.go [hunk 3]
```diff
@@ -292,7 +296,7 @@ func FromCLIConf(cf *CLIConf) (*BotConfig, error) {
 	}
 
 	if err := config.CheckAndSetDefaults(); err != nil {
-		return nil, trace.Wrap(err, "validing merged bot config")
+		return nil, trace.Wrap(err, "validating merged bot config")
```
Comment (timothyb89) [inline tool/tbot/config/config.go:299]: :facepalm:

**Review by zmb3 — APPROVED** [review]

**Review by timothyb89 — APPROVED** [review]

**Pull request merged**